src/components/Api.js

```diff
@@ -0,0 +1,174 @@
+export default class Api {
+  constructor(options) {
+    //constructor
+    this.baseUrl = options.baseUrl;
+    this._headers = options.headers;
+    //this._authorization = this.headers["authorization"];
+  }
+
+  getInitialCards(id) {
+    return fetch(`${this.baseUrl}/${id}`, {
+      headers: this._headers,
+    })
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        }
+
+        return Promise.reject(`Error: ${res.status}`);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+
+  getImageInfo(cardId) {
+    return fetch(`${this.baseUrl}/cards/${cardId}`, {
+      headers: this._headers,
+    }).then((res) => {
+      if (res.ok) {
+        return res.json();
+      }
+      return Promise.reject(`Error: ${res.status}`);
+    });
+  }
+
+  getUserInfo(path, id) {
+    return fetch(`${this.baseUrl}/${path}/${id}`, {
+      headers: this._headers,
+    })
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        }
+        return Promise.reject(`Error: ${res.status}`);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+
+  getPromiseAll() {
```

Comment:
This method should be called something like `getInitialData`, now the name is too general